lib/spack/spack/build_environment.py

```diff
@@ -789,6 +789,21 @@ def get_rpath_deps(pkg: spack.package_base.PackageBase) -> List[spack.spec.Spec]
     return _get_rpath_deps_from_spec(pkg.spec, pkg.transitive_rpaths)
 
 
+def get_rpaths(pkg):
```

Comment:
Transitive deps are covered by `CMAKE_INSTALL_RPATH_USE_LINK_PATH=ON`, so don't revive this. I removed it because it's inconsistent with whatever build_environment does, so bringing it back is a bug.

Reply:
Unfortunately, apply the suggested change is causing a failure in one of my scenario. I'll investigate tomorrow because it's not easy to state whether it's on my side or not since it involves hip support for which we have custom logic.

Reply:
OK I'm fine with things as they are now!